src/core/modules/Planet.cpp

```diff
@@ -4890,13 +4893,44 @@ void Planet::drawOrbit(const StelCore* core)
 
 	sPainter.setColor(getCurrentOrbitColor(), orbitFader.getInterstate());
 	Vec3d onscreen;
+
+	if (fromMoonPerspective) {
+		double dateJDE = lastJDE;
+		double calc_date;
+
+		Vec3d myPos = core->getCurrentPlanet()->getHeliocentricEclipticPos(dateJDE);
+		Vec3d myparentPos = getHeliocentricEclipticPos(dateJDE);
+		// orbital plane normal of parent planet, computed with the pos ~quarter of the orbit later
+		Vec3d orbitalNormal = myparentPos ^ getHeliocentricEclipticPos(dateJDE + deltaOrbitJDE*ORBIT_SEGMENTS/4);
+		orbitalNormal.normalize();
+		// project myPos into the orbital plane of parent planet using the normal of orbital plane
+		Vec3d myPosProjected = myPos - myPos.dot(orbitalNormal) * orbitalNormal;
+		// angle between myPos already projected onto the orbital plane of parent planet and the plane
+		// we should concentrate on this theta when sampling orbit
+		double theta = (atan2((myparentPos ^ myPosProjected).dot(orbitalNormal), myparentPos.dot(myPosProjected)))  / M_PI;
+
+		double f;
```

Comment:
This `f` is not used anywhere outside the loop, so should be defined there (as `const`, since it's never changed after the computation).

Reply:
fixed, thanks!